pilot/cmd/pilot-agent/app/cmd.go

```diff
@@ -311,6 +317,73 @@ func initProxy(args []string) (*model.Proxy, error) {
 	return proxy, nil
 }
 
+func applyExcludeInterfaces(ifaces []string) []string {
+	// Get list of excluded interfaces from pod annotation
+	annotations, err := bootstrap.ReadPodAnnotations("")
+	if err != nil {
+		log.Infof("Failed to read pod annotations to find exluded interfaces. Continuing without exclusions. Proxy IPAddresses: %v", ifaces)
+		return ifaces
+	}
+	value, ok := annotations[annotation.SidecarTrafficExcludeInterfaces.Name]
+	if !ok {
+		log.Infof("excludeInterfaces annotation not present. Proxy IPAddresses: %v", ifaces)
+		return ifaces
+	}
+	exclusions := strings.Split(value, ",")
+
+	// Find IP addr of excluded interfaces and add to a map for instant lookup
+	exclusionMap := make(map[string]struct{})
+	for _, ifaceName := range exclusions {
+		iface, err := net.InterfaceByName(ifaceName)
```

Comment:
nit: does it make sense to actually do the exclusion in getPrivateIPsIfAvailable directly?

Right now we list all interfaces, then get all the IPs (getPrivateIPsIfAvailable). Then in this function we go through the interfaces again and get the IPs and remove them from the list.

Seems more straightforward to just not add them in the first place?

Reply:
I thought about this and decided to go with a separate function instead of tainting the common utility function. It could be a surprise for the consumers who are not expecting any exclusions, if we modify getPrivateIPsIfAvailable.